File: benchmarks/pandas/bench_qcut.py
"""Benchmark: qcut (10 quantile bins) on 100k-element Series"""
import json, time
import numpy as np
import pandas as pd

ROWS = 100_000
WARMUP = 3
ITERATIONS = 10

data = (np.arange(ROWS) % 10000) * 0.01
s = pd.Series(data)

for _ in range(WARMUP):
    pd.qcut(s, 10, duplicates="drop")

start = time.perf_counter()
for _ in range(ITERATIONS):
    pd.qcut(s, 10, duplicates="drop")
total = (time.perf_counter() - start) * 1000

print(json.dumps({ "function": "qcut", "mean_ms": total / ITERATIONS, "iterations": ITERATIONS, "total_ms": total }))

File: benchmarks/tsb/bench_qcut.ts
/**
 * Benchmark: qcut (10 quantile bins) on 100k-element Series
 */
import { Series, qcut } from "../../src/index.js";

const ROWS = 100_000;
const WARMUP = 3;
const ITERATIONS = 10;

const data = Float64Array.from({ length: ROWS }, (_, i) => (i % 10000) * 0.01);
const s = new Series(data);

for (let i = 0; i < WARMUP; i++) {
  qcut(s, 10);
}

const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) {
  qcut(s, 10);
}
const total = performance.now() - start;

console.log(JSON.stringify({ function: "qcut", mean_ms: total / ITERATIONS, iterations: ITERATIONS, total_ms: total }));
